### ISSUE TITLE
Alternative worker pool for non-resource-consuming work actions

### ISSUE THREAD

**Issue by madisp** [gradle/gradle#25666]
### Expected Behavior

Ability to run work actions outside the normal worker pool bounded by max workers.

### Current Behavior (optional)

Today at least following the public APIs any workaction created from a gradle task consumes a worker from the worker pool bounded by max workers.

### Context

There are cases where a project is doing something in parallel but it doesn't actually consume local compute or IO resources. As an example, imagine running tests or deployment tasks on a remote machine.

Ideally these sorts of tasks wouldn't be bounded by the max workers setting as they don't consume any local resources but I found no such way today.

**Comment by ov7a** [comment]
Thank you for your interest in Gradle!

This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

**Comment by lptr** [comment]
This might be something we do at some point, but it's currently out of scope.

**Comment by ljacomet** [comment]
Since there might be interest in doing this one day, keeping it open for others to up-vote or relate specific issues to it.

There is however not commitment to do this in the near future.
